### PR TITLE
fix: the Response from DioError could return null from the data property but String is expected in ParseNetworkResponse

### PR DESCRIPTION
**Pull request by Nidal-Bakir**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues/774).

### Issue Description
The response from `dio` error object could have a null `data` property

Related issue: https://github.com/parse-community/Parse-SDK-Flutter/issues/774

### Approach
Add null aware operator `??`:
```
on dio.DioError catch (error) {
      return ParseNetworkByteResponse(
        data: error.response?.data ?? _fallbackErrorData,
        statusCode: error.response!.statusCode!,
      );
    }
```

### TODOs before merging
All set

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] A changelog entry

### CODE REVIEW

**Comment by mtrezza** [comment]
Could you please replace the PR title with `fix: <summary of what the issue is>`

**Review by mtrezza — CHANGES_REQUESTED** [review]
Could you add a test for this?

**Comment by Nidal-Bakir** [comment]
Sure, but I can not find a way to mock the `_ParseDioClient` class to fack the DioError

**Comment by mtrezza** [comment]
@parse-community/flutter-sdk-review what do you think, is this ready for merge?

**Comment by mtrezza** [comment]
@Nidal-Bakir any particular reason for closing this?

**Comment by mtrezza** [comment]
@parse-community/flutter-sdk-review should this PR be merged?

**Comment by Nidal-Bakir** [comment]
@mtrezza 
I'm sorry, I did not mean to close the PR! 
I did not know that if I deleted the forked repository from my account all my PR would be closed.

**Comment by mtrezza** [comment]
I've reopened, thanks for clarifying

File: packages/dart/lib/src/network/parse_dio_client.dart [hunk 1]
```diff
@@ -31,7 +31,9 @@ class ParseDioClient extends ParseClient {
           data: dioResponse.data!, statusCode: dioResponse.statusCode!);
     } on dio.DioError catch (error) {
       return ParseNetworkResponse(
-          data: error.response?.data, statusCode: error.response!.statusCode!);
+        data: error.response?.data ?? _fallbackErrorData,
+        statusCode: error.response!.statusCode!,
```
Comment (mbfakourii) [inline packages/dart/lib/src/network/parse_dio_client.dart:35]: I ran this code without internet connection and got null error in statusCode!
I suggest you change the code like this

```
return ParseNetworkResponse(
  data: error.response?.data ?? _fallbackErrorData,
  statusCode: error.response?.statusCode! ?? -1,
);
```

**Comment by Nidal-Bakir** [comment]
Reopening in a different PR because I lost all the progress and the reference of these changes [here](https://github.com/parse-community/Parse-SDK-Flutter/pull/825)

**Pull request closed without merging**